build/Dockerfile

```diff
@@ -1,22 +1,37 @@
 FROM registry.access.redhat.com/ubi7/ubi-minimal:latest
-ENV OPERATOR=/usr/local/bin/kubedirector
-# ENV DEBUGGER=/usr/local/bin/dlv
 
-RUN microdnf update -y && rm -rf /var/cache/yum
+ENV USER_UID=1001 \
+    USER_NAME=kubedirector \
+    USER_HOME=/home/kubedirector
 
+# Support ps and killall, for redeploy.
+RUN microdnf update -y && rm -rf /var/cache/yum
 RUN microdnf -y install --nodocs psmisc procps-ng \
     && microdnf clean all
 
-# Allow delve to run on Alpine based containers.
-# RUN apk add --no-cache libc6-compat
+# For OpenShift support, the runtime user will need to be able to self-insert
+# in /etc/passwd. We can't add the user during image build because on OpenShift
+# we'll get a random user ID at runtime (rather than USER_UID).
+RUN chmod g=u /etc/passwd
 
-# Debugger
-# COPY build/bin/dlv ${DEBUGGER}
-# RUN chmod +x ${DEBUGGER}
+# Install entrypoint script. This will create the user passwd entry and
+# launch the operator.
+COPY build/bin/entrypoint /usr/local/bin/entrypoint
+RUN chown 0:0 /usr/local/bin/entrypoint && \
```

Comment:
Why is the file owned by root instead of ${USER_UID} ?

Also, I know Swami already asked about this but can you elaborate why the group needs to be root?

Reply:
Just owned by root as a general pattern for the executables that live in /usr/local/bin. It's still broadly executable.

As for the root-group thing...

If the platform deploying this container honors the image's specified user & group, then it will be run with user ID 1001 and group ID 0. On OpenShift however (and perhaps other platforms with similar philosophy), the runner will have some other arbitrary/random user ID but still have group ID 0. Therefore we have to treat group ID as the important thing for accessing our operator-specific files. A little more about that here: https://docs.openshift.com/container-platform/3.11/creating_images/guidelines.html#openshift-specific-guidelines